cmake/compiler/gcc/target_base.cmake

```diff
@@ -12,3 +12,9 @@ macro(toolchain_cc_nocommon)
   zephyr_compile_options(-fno-common)
 
 endmacro()
+
+macro(toolchain_cc_cstd_flag dest_var_name c_std)
+
+  set_ifndef(${dest_var_name} "-std=${c_std}")
```

Comment:
And the convention for c_std is the regex "c[0-9]{2}".
So far I am only aware of two values "c99" and "c11" (from native_posix).

If someone's compiler requires the standard to be named with uppercase, they will have to remap it in their implementation of `toolchain_cc_cstd_flag`.